projects/gearbox/index.js

```diff
@@ -1,61 +1,158 @@
+    t.output,
+    t.input.target,
+  ]);
+
+  // Fetch ba;anes of tokens available to fetch
   const { output: totalBorrowed } = await sdk.api.abi.multiCall({
-    abi: abi['totalBorrowed'],
+    abi: abi["availableLiquidity"],
     calls: pools.map((pool) => ({ target: pool })),
     block,
-  })
+  });
+
+  const poolBalances = {};
+  totalBorrowed.forEach(({ output }, i) =>
+    sdk.util.sumSingleBalance(poolBalances, poolsUnderlying[i].output, output)
+  );
+
+  return { tokensAndOwners, poolBalances };
+};
+
+const getCreditManagers = async (block) => {
+  // Get DataCompressor from Gearbox addressProvider
+  const { output: DataCompressor } = await sdk.api.abi.call({
+    abi: abi["getDataCompressor"],
+    target: addressProvider,
+    block,
+  });
+  // Get gearbox pools from the contractsRegister, and underlyingToken for each pool
+  const { output: creditManagers } = await sdk.api.abi.call({
+    abi: abi["getCreditManagersList"],
+    target: DataCompressor,
+    block,
+  });
+
+  return creditManagers;
+};
 
-  const balances = {}
-  totalBorrowed.forEach(({ output }, i) => sdk.util.sumSingleBalance(balances, poolsUnderlying[i].output, output))
+const getV2TVL = async (block) => {
+  // Get Current CMs
+  const creditManagers = (await getCreditManagers(block)).filter(
+    (cm) => cm.version != 1
+  );
+  // Silently throw if no V2 CAs available
+  if (!creditManagers[0]) return [];
 
-  return sumTokens2({ balances, tokensAndOwners, block, })
-}
+  // Get all CA Balances
+  const caValues = await Promise.all(
+    creditManagers.map((cm) => getV2CAs(cm.creditFacade, block))
+  );
+
+  return creditManagers.map((cm, i) => ({
+    addr: cm.addr,
+    token: cm.underlying,
+    bal: caValues[i],
+  }));
+};
+
+const getV1TVL = async (block) => {
+  // Get Current CMs
+  const creditManagers = (await getCreditManagers(block)).filter(
+    (cm) => cm.version != 2
+  );
+
+  // Silently throw if no V2 CAs available
+  if (!creditManagers[0]) return [];
+
+  // Get all CA Balances
+  const caValues = await Promise.all(
+    creditManagers.map((cm) => getV1CAs(cm.addr, block))
+  );
+
+  return creditManagers.map((cm, i) => ({
+    addr: cm.addr,
+    token: cm.underlying,
+    bal: caValues[i],
+  }));
+};
+
+const tvl = async (timestamp, block) => {
+  // Pool TVL (Current token balances)
+  const { poolBalances, tokensAndOwners } = await getPoolAddrs(block);
+
+  // V2 CreditAccounts TVL in USD
+  const v2Balances = await getV2TVL(block);
+
+  // V1 CreditAccounts TVL
+  const v1Balances = await getV1TVL(block);
+
+  // Merge CA Values
+  const bal = {};
+  const tokens = [];
+  [...v2Balances, ...v1Balances].map((cm) => {
+    if (!bal[cm.token]) {
+      bal[cm.token] = cm.bal;
+      tokens.push([cm.token, cm.addr]);
+    } else {
+      bal[cm.token] = BigNumber.from(bal[cm.token]).add(cm.bal).toString();
+    }
+  });
+
+  return sumTokens2({
+    balances: { ...poolBalances, ...bal },
```

Comment:
the values from bal will overwrite token values in poolBalances, was this intentional? I have updated code to stop counting borrowed balances